torchseg/losses/dice.py

```diff
@@ -1,140 +1,147 @@
+            - eps: A small epsilon added to the denominator  for numerical stability to avoid nan
                 (denominator will be always greater or equal to eps)
 
         Shape
-             - **y_pred** - torch.Tensor of shape (N, C, H, W)
-             - **y_true** - torch.Tensor of shape (N, H, W) or (N, C, H, W)
+             - **y_pred** - torch.Tensor of shape (B, C, H, W),
+             - **y_true** - torch.Tensor of shape (B, C, H, W) or (B, 1, H, W),
+        where C is the number of classes.
 
         Reference
-            https://github.com/BloodAxe/pytorch-toolbelt
+            https://docs.monai.io/en/stable/_modules/monai/losses/dice.html#DiceLoss
         """
-        assert mode in {BINARY_MODE, MULTILABEL_MODE, MULTICLASS_MODE}
+        if reduction not in LossReduction.available_reductions():
+            raise ValueError(f'Unsupported reduction: {reduction}, '
+                             f'available options are {LossReduction.available_reductions()}.')
         super().__init__()
-        self.mode = mode
-        if classes is not None:
-            assert (
-                mode != BINARY_MODE
-            ), "Masking classes is not supported with mode=binary"
-            classes = to_tensor(classes, dtype=torch.long)
 
         self.classes = classes
         self.from_logits = from_logits
+        self.mask_to_one_hot = mask_to_one_hot
+        self.power = power
+        self.reduction = reduction
         self.smooth = smooth
         self.eps = eps
         self.log_loss = log_loss
         self.ignore_index = ignore_index
 
     def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor) -> torch.Tensor:
-        assert y_true.size(0) == y_pred.size(0)
-
-        if self.from_logits:
-            # Apply activations to get [0..1] class probabilities
-            # Using Log-Exp as this gives more numerically stable
-            # result and does not cause vanishing gradient on
-            # extreme values 0 and 1
-            if self.mode == MULTICLASS_MODE:
-                y_pred = y_pred.log_softmax(dim=1).exp()
-            else:
-                y_pred = F.logsigmoid(y_pred).exp()
 
-        bs = y_true.size(0)
-        num_classes = y_pred.size(1)
-        dims = (0, 2)
+        batch_size = y_pred.shape[0]
+        num_classes = y_pred.shape[1]
+        spatial_dims: list[int] = torch.arange(2, len(y_pred.shape)).tolist()
 
-        if self.mode == BINARY_MODE:
-            y_true = y_true.view(bs, 1, -1)
-            y_pred = y_pred.view(bs, 1, -1)
-
-            if self.ignore_index is not None:
-                mask = y_true != self.ignore_index
-                y_pred = y_pred * mask
-                y_true = y_true * mask
-
-        if self.mode == MULTICLASS_MODE:
-            y_true = y_true.view(bs, -1)
-            y_pred = y_pred.view(bs, num_classes, -1)
-
-            if self.ignore_index is not None:
```

Comment:
We are removing the ignore_index filtering but I don't see it added below in the new code